website/docs/10_going-deep/instrumentation.md

```diff
@@ -0,0 +1,138 @@
+# Booster instrumentation
+
+## Trace Decorator
+The Trace Decorator is a **Booster** functionality that facilitates the reception of notifications whenever significant events occur in Booster's core, such as event dispatching or migration execution.
+
+### Usage
+To configure a custom tracer, you need to define an object with two methods: onStart and onEnd. The onStart method is called before the traced method is invoked, and the onEnd method is called after the method completes. Both methods receive a TraceInfo object, which contains information about the traced method and its arguments.
+
+Here's an example of a custom tracer that logs trace events to the console:
+
+```typescript
+import {
+  TraceParameters,
+  BoosterConfig,
+  TraceActionTypes,
+} from '@boostercloud/framework-types'
+
+class MyTracer {
+  onStart(config: BoosterConfig, actionType: TraceActionTypes, traceParameters: TraceParameters): void {
+    console.log(`Start ${TraceActionTypes[actionType]}: ${traceParameters.className}.${traceParameters.methodName}`)
+  }
+
+  onEnd(config: BoosterConfig, actionType: TraceActionTypes, traceParameters: TraceParameters): void {
+    console.log(`End ${TraceActionTypes[actionType]}: ${traceParameters.className}.${traceParameters.methodName}`)
+  }
+}
+```
+
+You can then configure the tracer in your Booster application's configuration:
+
+```typescript
+import { BoosterConfig } from '@boostercloud/framework-types'
+import { MyTracer } from './my-tracer'
+
+const myTracer = new MyTracer()
+const config: BoosterConfig = {
+// ...other configuration options...
+  trace: {
+    enableTraceNotification: true,
+    verbose: false,
+    onStart: await myTracer.onStart,
+    onEnd: await myTracer.onStart,
+  }
+}
+```
+
+In the configuration above, we've enabled trace notifications and specified our onStart and onEnd as the methods to use. Verbose disable will reduce the amount of information generated excluding the internal parameter in the response. 
```

Comment:
Same here, we are talking about "Verbose". Maybe you wanted to say `includeInternal`? Also you say `excluding the internal parameter in the response`, What is the "internal parameter"? Also, I guess you wanted to say in the `trace parameters` not `in the response`, correct?

Anyway, I don't think there is a need to exclude information in the trace parameters. It can always be there, it doesn't hurt

Reply:
Yes, it's `includeInternal`. The `internal` parameter is in the documentation:

```
  internal: {
    target: unknown
    descriptor: PropertyDescriptor
  }
```

Regarding this parameter I think it's very useful as the methods (onStart and onEnd) are receiving an object with all the information. If you log the object (to console, file, etc...) and don’t exclude that property you may run into problems because of the size of the object. If the destination is a file you may end up having space problems for example. So it is very useful to give that option to the user.